modules/renderer/background_source.js

```diff
@@ -409,6 +405,37 @@ rendererBackgroundSource.Esri = function(data) {
             });
         }
 
+        // use a tilemap service to set maximum zoom for esri tiles dynamically
+        function fetchTilemap(center, esri) {
+            // tiles are available globally to zoom level 19, afterward they are only a possibility
+            const urlZ = 20;
+
+            // calculate url z/y/x from the lat/long of the center of the map
+            const urlX = (Math.floor((center[0] + 180) / 360 * Math.pow(2, urlZ)));
+            const urlY = (Math.floor((1 - Math.log(Math.tan(center[1] * Math.PI / 180) + 1 / Math.cos(center[1] * Math.PI / 180)) / Math.PI) / 2 * Math.pow(2, urlZ)));
+
+            // we fetch an 8x8 grid because they cover a normal extent and responses are cached
+            const tilemapUrl = tileCoord[3].replace(/tile\/[0-9]+\/[0-9]+\/[0-9]+/, 'tilemap') + `/${urlZ}/${urlY}/${urlX}/8/8`;
+
+            // make the request and introspect the response from the tilemap server
+            fetch(tilemapUrl)
+            .then(response => response.json())
```

Comment:
a bit of error handling might be good here... (url unavailable, cannot parse result, etc.)

Reply:
The bigger issue here is that we'll have to polyfill fetch / promises for IE11..
(But we need to do this anyway for D3 v5 / #4929)

Reply:
i don't have my heart set on using `fetch`, i just didn't know what you were using for AJAX requests to a CORS enabled web server because the only other import in the file previous was for making JSONP requests.

i've pushed up another commit which uses `d3_json` and aborts if an error is encountered. at this point i just need help triggering the function when editing is initially activated instead of waiting until the metadata context menu is opened.